src/components/example/certificate/AddComment.tsx

```diff
@@ -0,0 +1,89 @@
+import Button from '@/components/shared/button';
+import { useUserContext } from '@/contexts/UserContext';
+import { IComment } from '@/utils/types/certificate';
+import { FC, useState, ChangeEvent, useEffect } from 'react';
+
+type CommentProps = {
+  comments: IComment[] | undefined;
+  isCommentOpen: boolean;
+  setIsCommentOpen: (value: boolean) => void;
+};
+const AddComment: FC<CommentProps> = ({
+  comments,
+  isCommentOpen,
+  setIsCommentOpen,
+}) => {
+  const { currentUser } = useUserContext();
+  const [savedComments, setSavedComments] = useState<IComment[]>(
+    comments || [],
+  );
+  const [content, setContent] = useState<string>('');
+  const handleContentChange = (e: ChangeEvent<HTMLTextAreaElement>) => {
+    setContent(e.target.value);
+  };
+  useEffect(() => {
+    setSavedComments(comments || []);
```

Comment:
I am not sure if there is ever need to save as state something that was passed as prop. Something is either prop or it is state, it should not be both, then especially not set over effects. So please rework comment component so that comments are stored just on form and there is update function that adds comment to existing values on form.